### PR TITLE
Added --use-project-assets-json

### PR DESCRIPTION
**Pull request by jpeirson**
This addresses #29 by bypassing csproj file parsing and using the _project.assets.json_ file generated by nuget/dotnet restore.

* Adds `--use-project-assets-json`
* Fails if this is used without `-t` since the assets file includes all transitive refs. There might be a way to parse only immediate refs from the assets file, but I haven't looked into it. This is why I made this as a separate cmd line option rather than making it the default behavior.
* This option bypasses the .csproj parsing if any refs are found in the assets file.

### CODE REVIEW

**Comment by jpeirson** [comment]
I forgot to update the readme. Will do that in the next commit...

File: src/Methods.cs
Comment (tomchavakis) [inline src/Methods.cs:645]: Can you please add a unit test for this method?

**Comment by tomchavakis** [comment]
Hi @jpeirson,

I would like to thank you for this nice contribution. 
Can you please provide me with a scenario of using the project.assets.json? 
I am not familiar with this concept and I want to learn more.
Moreover, it would be great if you can add a unit test to cover this functionality (GetProjectReferencesFromAssetsFile).
What do you think?

**Comment by jpeirson** [comment]
I'll add a new commit with a unit test for the new method.

The primary scenario I had was the issue mentioned in #29 where some of my projects use a msbuild property as the package version, e.g.:
```
<PackageReference Include="MyApp.Lib1" Version="$(MyAppVersion)"/>
<PackageReference Include="MyApp.Lib2" Version="$(MyAppVersion)"/>
```

I'm running the `dotnet-project-licenses` command as part of my build process to generate a third-party licenses.txt file, which for me always needs to include the transitive refs in my first-party app.

Rather than trying to manually find and resolve that property, the msbuild system already handles all of the package reference resolution, including resolving any msbuild properties and finding transitive package refs, and the [msbuild system generates an .\obj\project.assets.json](https://kimsereyblog.blogspot.com/2018/08/sdk-style-project-and-projectassetsjson.html) for each SDK-style project or [otherwise any project that uses PackageReference](https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#using-packagereference-for-a-project-with-no-packagereferences) after running `dotnet restore` or `nuget restore` or the like. So the project.assets.json has all of the transitive packages without having to manually parse and crawl the project files for references.

I don't know if this approach works for older projects that use only packages.config, so I didn't want to break existing behavior, which is why I added a separate cmd line arg.

**Comment by jpeirson** [comment]
Added `GetProjectReferencesFromAssetsFile_Should_Resolve_Transitive_Assets`

File: src/Methods.cs
Comment (jpeirson) [reply src/Methods.cs:645]: Added `GetProjectReferencesFromAssetsFile_Should_Resolve_Transitive_Assets`

**Review by tomchavakis — APPROVED** [review]

**Comment by jpeirson** [comment]
Thanks @tomchavakis . I have green checks for approval, but I don't have write/merge access. If you want to merge, feel free to go ahead - I'm currently using the changes from this PR in our build process, so it looks good to me.

**Comment by tomchavakis** [comment]
Thanks, @jperson,
I will create a new release if you want to use it.

**Comment by jpeirson** [comment]
@tomchavakis thanks for creating the 2.3.2 release. I didn't bump up the version num in the csproj file, though, so the resulting nupkg is still 2.3.1. Should I have bumped up the version in csproj as part of this PR?

**Comment by tomchavakis** [comment]
@jpeirson,

I am sorry, I've just forgotten to update the .csproj version. I've pushed the new .csproj file and a new version will be ready in a few minutes.

**Comment by tomchavakis** [comment]
@jpeirson ,

versions v.2.3.3 just released.

**Pull request merged**
